presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergNativeMetadata.java

```diff
@@ -0,0 +1,523 @@
+        switch (name.getTableType()) {
+            case DATA:
+                break;
+            case HISTORY:
+                if (name.getSnapshotId().isPresent()) {
+                    throw new PrestoException(NOT_SUPPORTED, "Snapshot ID not supported for history table: " + systemTableName);
+                }
+                return Optional.of(new HistoryTable(systemTableName, table));
+            case SNAPSHOTS:
+                if (name.getSnapshotId().isPresent()) {
+                    throw new PrestoException(NOT_SUPPORTED, "Snapshot ID not supported for snapshots table: " + systemTableName);
+                }
+                return Optional.of(new SnapshotsTable(systemTableName, typeManager, table));
+            case PARTITIONS:
+                return Optional.of(new PartitionTable(systemTableName, typeManager, table, snapshotId));
+            case MANIFESTS:
+                return Optional.of(new ManifestsTable(systemTableName, table, snapshotId));
+            case FILES:
+                return Optional.of(new FilesTable(systemTableName, table, snapshotId, typeManager));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table)
+    {
+        return getTableMetadata(session, ((IcebergTableHandle) table).getSchemaTableName());
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaName)
+    {
+        if (schemaName.isPresent() && INFORMATION_SCHEMA.equals(schemaName.get())) {
+            return listSchemaNames(session).stream()
+                    .map(schema -> new SchemaTableName(INFORMATION_SCHEMA, schema))
+                    .collect(toList());
+        }
+
+        return resourceFactory.getCatalog(session).listTables(toIcebergNamespace(schemaName))
+                .stream()
+                .map(IcebergPrestoModelConverters::toPrestoSchemaTableName)
+                .collect(toList());
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        Table icebergTable = resourceFactory.getCatalog(session).loadTable(toIcebergTableIdentifier(table.getSchemaTableName()));
+        return getColumns(icebergTable.schema(), typeManager).stream()
+                .collect(toImmutableMap(IcebergColumnHandle::getName, identity()));
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        IcebergColumnHandle column = (IcebergColumnHandle) columnHandle;
+        return new ColumnMetadata(column.getName(), column.getType(), column.getComment().orElse(""), false);
```

Comment:
nit: hmmm, I'm not sure if we should set a null or use orElse("") here. why do you use an empty string rather than null?

Reply:
good point, this is to match the behavior of `IcebergMetadata`, but I agree it seems awkward to return empty string. That constructor is deprecated anyway, I will switch to use the builder.